### PR TITLE
Java: Connectors - Memory - Unit test coverage #1598

### PR DESCRIPTION
**Pull request by dsgrieve**
### Motivation and Context

Java: Connectors - Memory - Unit test coverage #1598 

### Description

Run unit tests with coverage and add additional tests to try to increase coverage. 
Added positive tests cases for AzureCognitiveSearchMemory. Negative tests and coverage need to be addressed.
Because this is somewhat time consuming, I suggest that be done in a separate PR rather than hold this PR. 
I have created [Java: Increase coverage of AzureCognitiveSearchMemoryTests ](https://github.com/microsoft/semantic-kernel/issues/1797) for that purpose.

### Contribution Checklist

- [ x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchMemory.java [hunk 1]
```diff
@@ -82,13 +82,13 @@ public AzureCognitiveSearchMemory(String endpoint, TokenCredential credentials)
                         .buildAsyncClient();
     }
 
-    private AzureCognitiveSearchMemory(AzureCognitiveSearchMemory other) {
-        this._adminClient = other._adminClient;
+    AzureCognitiveSearchMemory(SearchIndexAsyncClient adminClient) {
```
Comment (dsgrieve) [inline java/semantickernel-connectors-parent/semantickernel-connectors/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchMemory.java:85]: Note that this change was necessary for AzureCognitiveSearchMemoryTests

File: java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/MemoryRecordTests.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            _additionalMetadata);
+        MemoryRecord memoryRecord = new MemoryRecord(metadata, _embedding, "key", ZonedDateTime.now());
+
+        // Act
+        String serializedRecord = null;//JsonSerializer.Serialize(memoryRecord);
+        jsonString = jsonString.replaceAll("\n", "");
+        jsonString = jsonString.replaceAll(" ", "");
+
+        // Assert
+        assertEquals(jsonString, serializedRecord);
```
Comment (johnoliver) [inline java/semantickernel-core/src/test/java/com/microsoft/semantickernel/memory/MemoryRecordTests.java:237]: Think this should always fail, jsonString cant be null and serializedRecord must be null

**Comment by milderhc** [comment]
LGTM

**Review by milderhc — APPROVED** [review]

**Pull request merged**